music_assistant/providers/roku_media_assistant/player.py

```diff
@@ -0,0 +1,371 @@
+    from rokuecp import Roku
+
+    from .provider import MediaAssistantprovider
+
+
+class MediaAssistantPlayer(Player):
+    """MediaAssistantPlayer in Music Assistant."""
+
+    def __init__(
+        self,
+        provider: MediaAssistantprovider,
+        player_id: str,
+        roku_name: str,
+        roku: Roku,
+        queued: PlayerMedia | None = None,
+    ) -> None:
+        """Initialize the Player."""
+        super().__init__(provider, player_id)
+        # init some static variables
+        self.roku = roku
+        self.queued = queued
+        self._attr_name = roku_name
+        self._attr_type = PlayerType.PLAYER
+        self._attr_supported_features = {
+            PlayerFeature.POWER,  # if the player can be turned on/off
+            PlayerFeature.PAUSE,
+            PlayerFeature.VOLUME_MUTE,
+            PlayerFeature.ENQUEUE,
+        }
+        self._attr_volume_muted = False
+        self._attr_volume_level = 100
+
+        self.lock = asyncio.Lock()  # Held when connecting or disconnecting the device
+
+    async def setup(self) -> None:
+        """Set up player in MA."""
+        self._attr_available = False
+        self._attr_powered = False
+        await self.mass.players.register_or_update(self)
+
+    @property
+    def needs_poll(self) -> bool:
+        """Return if the player needs to be polled for state updates."""
+        return True
+
+    @property
+    def poll_interval(self) -> int:
+        """Return the interval in seconds to poll the player for state updates."""
+        return 5 if self.powered else 30
+
+    async def get_config_entries(self) -> list[ConfigEntry]:
+        """Return all (provider/player specific) Config Entries for the player."""
+        default_entries = await super().get_config_entries()
+        return [
+            *default_entries,
+            CONF_ENTRY_HTTP_PROFILE,
+        ]
+
+    async def power(self, powered: bool) -> None:
+        """Handle POWER command on the player."""
+        try:
+            device_info = await self.roku.update()
+
+            app_running = False
+
+            if device_info.app is not None:
+                app_running = device_info.app.app_id == self.provider.config.get_value(
+                    CONF_ROKU_APP_ID
+                )
+
+            # There's no real way to "Power" on the app since device wake up / app start
+            # is handled by The roku once it receives the Play Media request
+            if not powered:
+                self._attr_active_source = None
+                if app_running:
+                    await self.roku.remote("home")
+                    await self.roku.remote("power")
+
+            logger = self.provider.logger.getChild(self.player_id)
+            logger.info("Received POWER command on player %s", self.display_name)
+            # update the player state in the player manager
+            self.update_state()
+        except Exception:
```

Comment:
In a follow-up PR, try to narrow down this exception instead of broadly catching everything